Added support for StackTraceUsage With CallSite only (e.g. Caller Information attributes) + Improve CLScomplaint

snakefoot opened the pull request:
Extracted the callsite optimization from #2527, but excluded the special LoggerCallSite. And changed `enum StackTraceUsage` into a Flags-enum.

This allows NLog to make optimal use of caller-member-attributes to provide Callsite-information. Just need to figure out how to make a nice way to provide the info besides the fluent LogBuilder-interface.

Resolves partly #532 when using LogBuilder and resolves #2059

snakefoot commented:
Surprised by the special methods in `Log.cs`. I hope no one are using them as they must be crazy slow with disk-access for file-lookup on every LogEvent.

304NotModified commented on src/NLog/Config/StackTraceUsage.cs:
```diff
@@ -41,28 +41,38 @@ public enum StackTraceUsage
+        /// <summary>
+        /// Stack trace (with source) should only be captured if callsite details are missing
+        /// </summary>
+        WithCallSiteSource = 2,
```
please reoder the values. 

would it be better if we change this to a flags enum? e.g. WithCallSite | WithCallSiteSource

snakefoot replied:
StackTraceUsage is used everywhere as a level of details. The higher the number, the higher the detail. Not as a combination.

304NotModified replied:
good point, but what's higher? WithCallSiteSource  or WithoutSource ? ;)

snakefoot replied:
Also a very good point. Have now changed it into enum-flags.

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -279,17 +279,25 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
+            [CallerFilePath]
+#endif
+            string callerFilePath = null,
+#if NET4_5
```
maybe it was to long ago, but why only .net 4.5?

snakefoot replied:
Not sure I understand. NET4_5 means all newer versions (including NetStandard etc.)

304NotModified replied:
OK, but it wasn't there before. So why the change?

snakefoot replied:
Fixing  #2059 by just moving `NET4_5` (Has been there all the time)

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -302,21 +310,12 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
                 Property("CallerLineNumber", callerLineNumber);
 
-            _logEvent.SetCallerInfo(null, callerMemberName, callerFilePath, callerLineNumber);
+            if (callerMemberName != null || callerFilePath != null || callerLineNumber != 0)
```
if one of them isnot set, then we ignore them all?

it think `||` should be `&&`

snakefoot replied:
No it says only call `SetCallerInfo` if one of them is NOT null. Better handle NetFramework version without support for caller-member-attributes (NET3_5 +NET4_0)

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -367,38 +363,35 @@ public void WriteIf(Func<bool> condition)
                 Property("CallerLineNumber", callerLineNumber);
 
-            _logEvent.SetCallerInfo(null, callerMemberName, callerFilePath, callerLineNumber);
+            if (callerMemberName != null || callerFilePath != null || callerLineNumber != 0)
```
please extract a method for this.

snakefoot replied:
Done

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -367,38 +363,35 @@ public void WriteIf(Func<bool> condition)
             if (!condition || !_logger.IsEnabled(_logEvent.Level))
                 return;
 
+            // TODO NLog 5.0 Remove these properties
```
this is for NLog 5? So we could remove this?

snakefoot replied:
Well it is todo work. But I would like to do it in separate commit. Like to keep things apart, since removing these properties requires cleanup other places.

304NotModified replied:
OK 👍

304NotModified commented on src/NLog/Internal/TargetWithFilterChain.cs:
```diff
@@ -50,6 +51,115 @@ internal class TargetWithFilterChain
+            /// <returns>True if objects are equal, false otherwise.</returns>
+            public bool Equals(CallSiteKey other)
+            {
+                return FileSourceLineNumber == other.FileSourceLineNumber
```
I think it's missing the null comparisons, See https://docs.microsoft.com/en-us/previous-versions/ms173147(v=vs.90)

PS: Resharper could generate equals and hashcode

snakefoot replied:
CallSiteKey is a struct, so no need for null-check

304NotModified replied:
ah missed that, thanks

304NotModified commented on src/NLog/Internal/TargetWithFilterChain.cs:
```diff
@@ -50,6 +51,115 @@ internal class TargetWithFilterChain
+               )
+                return false;
+
+            if (logEvent.CallSiteInformation == null)
```
we prefer to combine this with below, `  if (string.IsNullOrEmpty(logEvent.CallSiteInformation?.CallerFilePath))`

snakefoot replied:
Done

304NotModified commented on src/NLog/Internal/TargetWithFilterChain.cs:
```diff
@@ -50,6 +51,115 @@ internal class TargetWithFilterChain
+            string internClassName = logEvent.LoggerName == className ?
+                logEvent.LoggerName :
+#if !NETSTANDARD1_0
+                string.Intern(className);
```
why do we need this? Please add some code comments, as interning is really not necessary is most applications

snakefoot replied:
Added comment. But the classname-extractor can perform substring, and other logic. Want to avoid allocating class-name-strings for every logging-location in a source-file.

304NotModified commented on src/NLog/Internal/TargetWithFilterChain.cs:
```diff
@@ -50,6 +51,115 @@ internal class TargetWithFilterChain
+            if (string.IsNullOrEmpty(className))
+                return false;
+
+            if (_callSiteClassNameCache == null)
```
I think this check should be earlier in this method (cheaper check)

snakefoot replied:
Well this is just a safe-guard in case one has called in an expected order. So it will "never" catch anything.

304NotModified commented on src/NLog/Internal/TargetWithFilterChain.cs:
```diff
@@ -50,6 +51,115 @@ internal class TargetWithFilterChain
+            CallSiteKey callSiteKey = new CallSiteKey(logEvent.CallerMemberName, logEvent.CallerFilePath, logEvent.CallerLineNumber);
+            return _callSiteClassNameCache.TryGetValue(callSiteKey, out callSiteClassName);
+        }
+        private MruCache<CallSiteKey, string> _callSiteClassNameCache;
```
please add fields as first in the class

snakefoot replied:
Done

304NotModified commented on src/NLog/LayoutRenderers/CallSite and stacktraces/CallSiteLayoutRenderer.cs:
```diff
@@ -130,12 +130,10 @@ StackTraceUsage IUsesStackTrace.StackTraceUsage
 #endif
-
-                return StackTraceUsage.WithoutSource;
+                    return SkipFrames == 0 ? StackTraceUsage.WithCallSite : StackTraceUsage.WithoutSource;
```
please create a helper for this

304NotModified replied:
FYI,

I'm trying to follow https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) these days in NLog

snakefoot replied:
Done

304NotModified commented:
> Surprised by the special methods in `Log.cs`. I hope no one are using them as they must be crazy slow with disk-access for file-lookup on every LogEvent.

Do you mean Path.GetFileNameWithoutExtension(callerFilePath)?

snakefoot commented:
Ah yes sorry. Thought it tried to find the filename on the disk. But it is "just' removing the file-extension. Not so bad besides doing string allocation and Logger-lookup on every logevent


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Surprised by the special methods in Log.cs. I hope no one are using them as they must be crazy slow with disk-access for file-lookup on every LogEvent.

Do you mean Path.GetFileNameWithoutExtension(callerFilePath)?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HGP7E573HOHBCR7QT3PX32HRA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWQXGBY#issuecomment-497120007>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HEGND2TZF7OFRB5HCDPX32HRANCNFSM4HPVH35A>.

snakefoot commented:
Going on a 10 day vacation. So if more changes are needed before merge, then you are welcome to make commit to the branch :)

304NotModified commented:
Have a good holiday!

304NotModified commented on src/NLog/Config/StackTraceUsage.cs:
```diff
@@ -33,36 +33,56 @@
         /// Stack trace should be captured without source-level information.
         /// </summary>
-        WithoutSource = 1,
+        WithoutSource = WithStackTrace,
```
what's the use of this alias?

304NotModified replied:
backwards comp 👍

304NotModified replied:
maybe we should `[Obsolete]` this one.

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -279,44 +279,34 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
+            [CallerFilePath]
+#endif
+            string callerFilePath = null,
+#if NET4_5
```
I'm still a bit confused by this :(

In .net 4, why having this method with 3 parameters? What am I missing?

snakefoot replied:
Make Net45 and Net40 binary compatible.

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -279,44 +279,34 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
-        /// <summary>
-        /// Writes the log event to the underlying logger.
-        /// </summary>
-        public void Write()
```
maybe i'm missing it, but I think this was better

304NotModified replied:
If possible, please revert to this

snakefoot replied:
Have partially reverted so your eyes will relax.

304NotModified commented on src/NLog/LoggerImpl.cs:
```diff
@@ -55,19 +55,35 @@ internal static void Write([NotNull] Type loggerType, [NotNull] TargetWithFilter
+                bool attemptCallSiteOptimization = targetsForLevel.TryCallSiteClassNameOptimization(stu, logEvent);
+                if (attemptCallSiteOptimization && targetsForLevel.TryLookupCallSiteClassName(logEvent, out string callSiteClassName))
+                {
+                    logEvent.CallSiteInformation.CallerClassName = callSiteClassName;
```
shouldn't we check ` logEvent.CallSiteInformation` for null?

304NotModified replied:
or should we use `GetCallSiteInformationInternal()`?

snakefoot replied:
There is no need for null check as it needs the callerinfo-details to make the lookup (details equals null means no lookup and no assignment of class-name)

304NotModified commented on src/NLog/Targets/Target.cs:
```diff
@@ -521,9 +521,9 @@ private void FindAllLayouts()
-            StackTraceUsage = _allLayouts.DefaultIfEmpty().Max(layout => layout?.StackTraceUsage ?? StackTraceUsage.None);
-            if (this is IUsesStackTrace usesStackTrace && usesStackTrace.StackTraceUsage > StackTraceUsage)
-                StackTraceUsage = usesStackTrace.StackTraceUsage;
+            StackTraceUsage = _allLayouts.DefaultIfEmpty().Aggregate(StackTraceUsage.None, (seed,layout) => seed | (layout?.StackTraceUsage ?? StackTraceUsage.None));
```
I refactored the Aggregate, those are hard to read

snakefoot replied:
Thought you hated my for-loops :)

snakefoot commented:
Replying from my phone. The moving of NET4_5 for caller member attributes. Is to make NET4_0 + NET3_5 binary compatible. One could also add 3 default parameters to the now removed Write-method with the exact same parameters name.



I have not removed the old enum names in attenpt to make it less breaking. But you are welcome to remove all the old aliases.



There is no need for null check as it needs the callerinfo-details to make the lookup (details equals null means no lookup and no assignment of class-name)




Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Have a good holiday!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HED4T5RHUGM4OFHNLTPX667PA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWSEWZA#issuecomment-497306468>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HDHICWAH63JDHIMMY3PX667PANCNFSM4HPVH35A>.

snakefoot commented:
Replying from my phone. The moving of NET4_5 for caller member attributes. Is to make NET4_0 + NET3_5 binary compatible. One could also add 3 default parameters to the now removed Write-method with the exact same parameters name.


I have not removed the old enum names in attenpt to make it less breaking. But you are welcome to remove all the old aliases.


There is no need for null check as it needs the callerinfo-details to make the lookup (details equals null means no lookup and no assignment of class-name)



Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


@304NotModified commented on this pull request.

________________________________

In src/NLog/Config/StackTraceUsage.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289130220>:

>
         /// <summary>
         /// Stack trace should be captured without source-level information.
         /// </summary>
-        WithoutSource = 1,
+        WithoutSource = WithStackTrace,


what's the use of this alias?

________________________________

In src/NLog/Fluent/LogBuilder.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289131193>:

>          /// <summary>
         /// Writes the log event to the underlying logger.
         /// </summary>
         /// <param name="callerMemberName">The method or property name of the caller to the method. This is set at by the compiler.</param>
         /// <param name="callerFilePath">The full path of the source file that contains the caller. This is set at by the compiler.</param>
         /// <param name="callerLineNumber">The line number in the source file at which the method is called. This is set at by the compiler.</param>
         public void Write(
-            [CallerMemberName]string callerMemberName = null,
-            [CallerFilePath]string callerFilePath = null,
-            [CallerLineNumber]int callerLineNumber = 0)
+#if NET4_5
+            [CallerMemberName]
+#endif
+            string callerMemberName = null,
+#if NET4_5
+            [CallerFilePath]
+#endif
+            string callerFilePath = null,
+#if NET4_5


I'm still a bit confused by this :(

In .net 4, why having this method with 3 parameters? What am I missing?

________________________________

In src/NLog/Fluent/LogBuilder.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289131348>:

>
             _logger.Log(_logEvent);
         }
-#else
-        /// <summary>
-        /// Writes the log event to the underlying logger.
-        /// </summary>
-        public void Write()


maybe i'm missing it, but I think this was better

________________________________

In src/NLog/LoggerImpl.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289132775>:

>              {
+                bool attemptCallSiteOptimization = targetsForLevel.TryCallSiteClassNameOptimization(stu, logEvent);
+                if (attemptCallSiteOptimization && targetsForLevel.TryLookupCallSiteClassName(logEvent, out string callSiteClassName))
+                {
+                    logEvent.CallSiteInformation.CallerClassName = callSiteClassName;


shouldn't we check logEvent.CallSiteInformation for null?

________________________________

In src/NLog/LoggerImpl.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289133127>:

>              {
+                bool attemptCallSiteOptimization = targetsForLevel.TryCallSiteClassNameOptimization(stu, logEvent);
+                if (attemptCallSiteOptimization && targetsForLevel.TryLookupCallSiteClassName(logEvent, out string callSiteClassName))
+                {
+                    logEvent.CallSiteInformation.CallerClassName = callSiteClassName;


or should we use GetCallSiteInformationInternal()?

________________________________

In src/NLog/Targets/Target.cs<https://github.com/NLog/NLog/pull/3431#discussion_r289134461>:

> @@ -521,9 +521,9 @@ private void FindAllLayouts()
             {
                 _allLayoutsAreThreadSafe = _allLayouts.All(layout => layout.ThreadSafe);
             }
-            StackTraceUsage = _allLayouts.DefaultIfEmpty().Max(layout => layout?.StackTraceUsage ?? StackTraceUsage.None);
-            if (this is IUsesStackTrace usesStackTrace && usesStackTrace.StackTraceUsage > StackTraceUsage)
-                StackTraceUsage = usesStackTrace.StackTraceUsage;
+            StackTraceUsage = _allLayouts.DefaultIfEmpty().Aggregate(StackTraceUsage.None, (seed,layout) => seed | (layout?.StackTraceUsage ?? StackTraceUsage.None));


I refactored the Aggregate, those are hard to read

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HHZPHD3OJJ3S2FIBPTPYASITA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2FMIXA#pullrequestreview-243975260>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HBC6RXXA6DD36ZNSZTPYASITANCNFSM4HPVH35A>.

304NotModified commented:
Thanks for the reply!

snakefoot commented:
Btw found this page where the wiki is used for  issue tracking


https://github.com/NLog/NLog/wiki/Unable-to-write-to-NLOG-in-kubernetes-containers


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Thanks for the reply!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HBCPVWQTEO5LUHTORLPYGO2LA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWWPTMI#issuecomment-497875377>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HAAWLNX3OAMORNVVVDPYGO2LANCNFSM4HPVH35A>.

304NotModified commented:
converted this to an issue -> https://github.com/NLog/NLog/issues/3446

snakefoot commented:
Regarding the new issue then absolute file path should be used that matches the mounted volumes configured. Kubernetes write permissions should of course be given.


Else good idea to upgrade to latest NLog and check the internal logging (maybe redirect to console if that is working)


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


converted this to an issue -> #3446<https://github.com/NLog/NLog/issues/3446>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HFOQEH2XC3NPXTA563PYQOETA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWX4N2A#issuecomment-498058984>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HA57H637BAMYEIDHATPYQOETANCNFSM4HPVH35A>.

304NotModified commented:
> Regarding the new issue then absolute file path should be used that matches the mounted volumes configured. Kubernetes write permissions should of course be given. Else good idea to upgrade to latest NLog and check the internal logging (maybe redirect to console if that is working)

Is this related to this PR?

snakefoot commented:
Nope to the issue created from the rogue wiki page


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Regarding the new issue then absolute file path should be used that matches the mounted volumes configured. Kubernetes write permissions should of course be given. Else good idea to upgrade to latest NLog and check the internal logging (maybe redirect to console if that is working)

Is this related to this PR?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/NLog/NLog/pull/3431?email_source=notifications&email_token=ACXZ7HDTI4RUXPKYJQ2I6Y3PYU7JTA5CNFSM4HPVH35KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWZ5CNQ#issuecomment-498323766>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACXZ7HESNFXVDKUFIQHYN63PYU7JTANCNFSM4HPVH35A>.

304NotModified commented on src/NLog/Config/StackTraceUsage.cs:
```diff
@@ -33,36 +33,56 @@
+        /// <summary>
+        /// Source-level information should be capture (source-filename + linenumber)
+        /// </summary>
+        WithSourceCode = 2,
```
The big reason why this isn't merged yet are the changed write method and the names of this enum.

I'm doubting if sourcecode is clear. But for now haven't better names

304NotModified replied:
Maybe rename this to WithFileNameAndLineNumber?

snakefoot replied:
Fine with me

304NotModified commented on src/NLog/Config/StackTraceUsage.cs:
```diff
@@ -33,36 +33,56 @@
+        /// <summary>
+        /// Stack trace should be captured
+        /// </summary>
+        WithStackTrace = 1,
```
maybe: WithFullStackTrace?

snakefoot replied:
What is the difference between "full stacktrace" and "stacktrace" ?

304NotModified replied:
it's to make the difference between stacktrace and callsite more clear. But OK, this isn't a good proposal

304NotModified commented on src/NLog/Config/StackTraceUsage.cs:
```diff
@@ -33,36 +33,56 @@
+        /// <summary>
+        /// Stack trace should only be captured if callsite details are missing
+        /// </summary>
+        WithCallSite = 4,
```
maybe WithOnlyCallSite?

snakefoot replied:
Please tell reason

304NotModified replied:
I've difficulties to understand the difference between all the options now.  I'm afraid it's too difficult for the users.

snakefoot replied:
Only highly skilled people who knows about the NLog Callsite-Engine and will be creating their own special LayoutRenderers/Targets will have to know about these enum-values. Pretty sure no-one knows how to use them right now.

304NotModified replied:
But maybe that should be in the docs. It isn't explaining now what it does (and so the difference between callsite and stackstrace is difficult)

Please note that in VS there is also a callstack window, which is 50% callsite  and 50% stackstrace  ;)

snakefoot replied:
Well in NLog callsite means the calling method (aka caller information). and stacktrace means the callstack. Both of them have the option to include-source-information (filename + linenumber). But I'm pretty sure you know all this already.

304NotModified replied:
When trying to keep things simple, I'm not the only audience ;)

snakefoot replied:
NLog callsite-name comes from the `${callsite}` layoutrenderer. Keeping this name because I guess people using NLog, are used to the NLog-names.

snakefoot replied:
> When trying to keep things simple, I'm not the only audience ;)

Well I guess there 4 people in total, when including me and you :)

snakefoot replied:
I find the enum-names very simple. And have just kept the old enum-names to ensure backward-compatible. But made it very explicit that they are combination of the new enum-flag-values.

304NotModified replied:
Updated docs, this is fine for me know. Thanks!

304NotModified commented on src/NLog/Fluent/LogBuilder.cs:
```diff
@@ -294,36 +294,28 @@ public LogBuilder StackTrace(StackTrace stackTrace, int userStackFrame)
+        public void Write(
+            string callerMemberName = null,
+            string callerFilePath = null,
+            int callerLineNumber = 0)
```
I don't fully understand why the signature changed for `Write` for < net 45? 

(the docs outside is a good idea)

snakefoot replied:
I want to make the Logger-interface binary compatible.

304NotModified replied:
While that's a nice property, it's not really useful for now? We can't (runtime) replace the nlog.dll for .net4 with .net45 already. 

I prefer revert of this as it could confuse users. (the optional parameters are visible)

snakefoot replied:
Not sure I understand the issue with confused users. Net45 users are also looking at default parameters.

If you have a net40 library compiled with NLog Net40 that uses fluent-interface, and then you attempt to use that library in an Net45 application, then you will get method-not-found. This is the issue I'm trying to fix. The NLog Logger-interface should not give such surprises.

304NotModified replied:
> This is the issue I'm trying to fix

ah I would expect more issues, but OK you convinced me.  :)

304NotModified commented on src/NLog/LogEventInfo.cs:
```diff
@@ -204,7 +204,13 @@ public LogEventInfo(LogLevel level, string loggerName, IFormatProvider formatPro
         /// </summary>
-        public string CallerMemberName => CallSiteInformation?.GetCallerMemberName(null, false, true, true);
+        [Obsolete("Use CallerMethodName instead. Marked obsolete on NLog 5.0")]
+        public string CallerMemberName => CallSiteInformation?.GetCallerMethodName(null, false, true, true);
```
Why the rename? The attribute is called CallerMemberName

Maybe method is better , but I really don't prefer obsoletes

snakefoot replied:
The rename is because I find "CallerMethodName" easier to understand. And there are tools like Fody that can also be used to capture caller-context, where they don't use the special attributes.

304NotModified replied:
Clear. But because of the obsolete I think we should not do this. It's not that important IMO and obsoletes should be prevented.

So please revert/remove the commit

snakefoot replied:
Have reverted the rename of CallerMemberName

304NotModified replied:
Thanks!

304NotModified commented on src/NLog/LogEventInfo.cs:
```diff
@@ -551,12 +551,12 @@ public void SetStackTrace(StackTrace stackTrace, int userStackFrame)
         /// <param name="callerFilePath"></param>
         /// <param name="callerLineNumber"></param>
-        public void SetCallerInfo(string callerClassName, string callerMemberName, string callerFilePath, int callerLineNumber)
+        public void SetCallerInfo(string callerClassName, string callerMethodName, string callerFilePath, int callerLineNumber)
```
I'm afraid this is confusing. Please membername everywhere. Using 2 names for one thing is confusing.

snakefoot replied:
Fixed

304NotModified replied:
Is still see many callerMethodNames?

snakefoot replied:
Thought you agreed that `callerMemberName` was a confusing name for something that is a method-name. See no reason for a badly chosen property-name to obfuscate the rest of the internal code (But leaving the smell in the public interface). Notice that when extracting the method-name from stacktrace, then it is correctly called method-name.

304NotModified commented:
Sorry for the long delay!
